### ISSUE TITLE
Move GeoInterace methods to an extension in GeoInterface to remove dependency loops

### ISSUE THREAD

**Issue by rafaqz** [JuliaGeometry/GeometryBasics.jl#221]
As MakieCore depends on GeometryBasics there is no way for GeoInterface to add makie recipes without a dependency loop.

We want the GeoInterface geometries to plot easily, and for our macro to add Makie recipes to be useable from GeoInterface. 

The problem is going to be managing the transition without breaking lots of code and / or precompile.

**Comment by asinghvi17** [comment]
Can you call `geointerface_geomtype` methods from an extension?  Those at least need GI traits for dispatch...

**Comment by rafaqz** [comment]
Good point. We can shift to module dispatch in GI.convert so `geointerface_geomtype` isn't hit.

That's the plan anyway with 1.10 being LTS

**Comment by asinghvi17** [comment]
Ah, yeah the current method could be kept as a fallback.
